tests/integration/test_social_media_services.py

```diff
@@ -0,0 +1,112 @@
+# Standard Stuff
+import json
+
+# Third Party Stuff
+import pytest
+import tweepy
+from unittest import mock
+
+# nexus Stuff
+from tests import utils as u
+from django.urls import reverse
+from django.utils import timezone
+from tests import factories as f
+from nexus.social_media import services
+from nexus.social_media.models import Post
+from settings.development import TWITTER_OAUTH
+from nexus.base.exceptions import WrongArguments
+from django.core.files.uploadedfile import SimpleUploadedFile
+
+
+pytestmark = pytest.mark.django_db
+
+
+def test_update_post_object(client):
+    user = f.create_user(email='test@example.com', password='test')
+    client.login(user=user)
+
+    url = reverse('posts-list')
+    post = {
+        'text': 'Announcement!',
+        'posted_at': 'twitter',
+        'scheduled_time': '2018-10-09T18:30:00Z'
+    }
+
+    response = client.json.post(url, json.dumps(post))
+    post_instance = Post.objects.get(pk=response.data['id'])
+
+    assert post_instance.is_posted is False
+    assert post_instance.posted_time is None
+
+    services.update_post_object(post_instance)
```

Comment:
You should check the API here. I'm not sure what you're trying to do after this line.

Reply:
Okay, so once the post is published there is a need to update two of its arguments `is_posted=True` and `posted_time=timezone.now()` and then saving this updated `Post` instance to the database. For this complete process, I created a service `update_post_object` that is called if the post is published to Twitter without raising any `TweepError`.

After this line in this test, I am just testing my service `update_post_object`. Checking, if a `Post` instance is passed to it that has `is_posted==False` and `posted_time==None`, then does this service update required arguments or not?
Thus, after this line I am checking:
`assert post_instance.is_posted is True` and
`assert post_instance.posted_time.date() == timezone.now().date()`

I think a simpler version of this second assertion would be `assert post_instance.posted_time is not None`.

Reply:
This will be flaky: `post_instance.posted_time.date() == timezone.now().date()`

What happens when your update post is called at 11:59:59 and comes back in the test at 00:00:00?

The date will change no matter whatever timezone you're in.